packages/did/src/Did.utils.ts

```diff
@@ -263,8 +264,53 @@ export function isVerificationKey(
 export function isEncryptionKey(
   key: Partial<NewDidKey | DidKey> & Pick<NewDidKey | DidKey, 'type'>
 ): key is NewDidEncryptionKey | DidEncryptionKey {
-  const keyType = key.type.toLowerCase()
-  return encryptionKeyTypes.some((kt) => kt === keyType)
+  return encryptionKeyTypes.some((kt) => kt === key.type)
+}
+
+export const verificationKeyTypeToChainKeyType: Record<
+  VerificationKeyType,
+  DidDidDetailsDidVerificationKey['type']
+> = {
+  sr25519: 'Sr25519',
+  ed25519: 'Ed25519',
+  ecdsa: 'Ecdsa',
+}
+
+export const chainVerificationKeyTypeToKeyType = Object.entries(
+  verificationKeyTypeToChainKeyType
+).reduce((obj, [key, value]) => ({ ...obj, [value]: key }), {}) as Record<
+  DidDidDetailsDidVerificationKey['type'],
+  VerificationKeyType
+>
+
+export type EncodedVerificationKey =
+  | { Sr25519: Uint8Array }
+  | { Ed25519: Uint8Array }
+  | { Ecdsa: Uint8Array }
+
+export const encryptionKeyTypeToChainKeyType: Record<
+  EncryptionKeyType,
+  DidDidDetailsDidEncryptionKey['type']
+> = {
+  x25519: 'X25519',
+}
+
+export const chainEncryptionKeyTypeToKeyType = Object.entries(
+  encryptionKeyTypeToChainKeyType
+).reduce((obj, [key, value]) => ({ ...obj, [value]: key }), {}) as Record<
+  DidDidDetailsDidEncryptionKey['type'],
+  EncryptionKeyType
+>
+
+export type EncodedEncryptionKey = { X25519: Uint8Array }
+
+export type EncodedKey = EncodedVerificationKey | EncodedEncryptionKey
+
+export type EncodedSignature = EncodedVerificationKey
```

Comment:
Since I suggest removing the generic TypedValue we get a tiny bit of duplication, which I am OK to live with. Makes things explicit as well.